platform/mds/mds-web/src/main/resources/webapp/js/directives.js

```diff
@@ -2189,6 +2189,47 @@
         };
     });
 
+    directives.directive('entityInstanceFields2', function (Entities) {
+        return {
+            restrict: 'AE',
+            transclude: true,
+            scope: true,
+            controller: ['$scope', function ($scope) {
+                $scope.subclassCurrent = $scope.$parent.$parent.$parent.subclassCurrent;
+                console.log("rendering subclassCurrent: ",$scope.subclassCurrent);
+            }],
+            templateUrl: '../mds/resources/partials/widgets/entityInstanceFields.html'
+        };
+    });
+
+    directives.directive('entityInstanceField', function (Entities) {
+        return {
+            restrict: 'AE',
+            scope: true,
+            controller: ['$scope', '$timeout', function ($scope, $timeout) {
+                if($scope.field.name === 'subclass') {
+                    $scope.$watch(
+                        function() {return $scope.field.value;},
+                        function(valueNew, valueOld) {
+                        if(valueNew) {
+                            $timeout(function() {
+                                // TODO: six parents is poor coupling with the template layouts.
+                                // I need to carefully review the recently written docs for reuse
+                                // of the data browser https://github.com/motech/motech/pull/100
+                                // and see if I can do something stylistically compatible, while not
+                                // giving up my plan to make the data browser be able to recurseively
+                                // display related instances.
+                                $scope.$parent.$parent.$parent.$parent.$parent.$parent.subclassCurrent = valueNew;
```

Comment:
As mentioned, we have a method for finding the scope, maybe use that. But from what I can see we only to do this for the 'draft' method, I don't see us doing anything like this, but I might be missing something. Why do you need to go so far up this chain? Maybe there is a better way to pass these values?